File: Graphs/TreeGraphs/examples/tree_graph_example.py
from Graphs.TreeGraphs.source.tree_graph import TreeGraph

if __name__ == "__main__":
    # Create a tree graph with root "A"
    tree = TreeGraph("A")

    # Add nodes to the tree
    tree.add_node("A", "B")
    tree.add_node("A", "C")
    tree.add_node("B", "D")
    tree.add_node("B", "E")
    tree.add_node("C", "F")
    tree.add_node("F", "G")

    # Display the tree structure
    print("Tree Structure:")
    tree.display()

    # Perform different tree traversals
    print("\nIn-order Traversal:")
    tree.traversal_in_order()

    print("\n\nPre-order Traversal:")
    tree.traversal_pre_order()

    print("\n\nPost-order Traversal:")
    tree.traverse_post_order()
    print("\n")

File: Graphs/TreeGraphs/examples/tree_graph_example.js
const TreeGraph = require("../source/tree_graph");

// Create an instance of the TreeGraph with a root node
const tree = new TreeGraph("A");

// Add nodes to the tree
tree.addNode("A", "B");
tree.addNode("A", "C");
tree.addNode("B", "D");
tree.addNode("B", "E");
tree.addNode("C", "F");
tree.addNode("F", "G");

// Display the tree structure
console.log("Tree Structure:");
tree.display();

// Perform in-order traversal
console.log("\nIn-order Traversal:");
tree.traversalInOrder();

// Perform pre-order traversal
console.log("\n\nPre-order Traversal:");
tree.traversalPreOrder();

// Perform post-order traversal
console.log("\n\nPost-order Traversal:");
tree.traversalPostOrder();
